src/PHPUnit/Controller/AbstractHttpControllerTestCase.php

```diff
@@ -74,13 +74,13 @@ public function assertHasResponseHeader($header)
     public function assertNotHasResponseHeader($header)
     {
         $responseHeader = $this->getResponseHeader($header);
-        if (false !== $responseHeader) {
+        if ($responseHeader) {
             throw new ExpectationFailedException($this->createFailureMessage(sprintf(
                 'Failed asserting response header "%s" WAS NOT found',
                 $header
             )));
         }
-        $this->assertFalse($responseHeader);
+        $this->assertEquals(false, $responseHeader);
```

Comment:
Few things wrong here:

 1. `assertFalse()` would give the best error message here
 2. `assertEquals` would compare `false` and `''`, and consider them equal
 3. `assertEquals` would give us a confusing error message